framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -2429,69 +2429,84 @@ private static final boolean validType(TypeMirror type) {
         }
     }
 
-    /** Sets indexTypes and indexDeclAnnos by side effect, just before returning. */
-    protected void buildIndexTypes() {
-        if (this.indexTypes != null || this.indexDeclAnnos != null) {
-            ErrorReporter.errorAbort("AnnotatedTypeFactory.buildIndexTypes called more than once");
+    /**
+     * Parses the stub files in the following order: <br>
+     * 1. jdk.astub in the same directory as the checker, if it exists and ignorejdkastub option is not supplied <br>
+     * 2. flow.astub in the same directory as BaseTypeChecker <br>
+     * 3. Stub files listed in @Stubfiles annotation on the checker; must be in same directory as the checker<br?
+     * 4. Stub files provide via stubs compiler option <br>
```

Comment:
Should we change this order to be 6 - 5 - 4 ?
6 vs 5 I don't care so much. But shouldn't the individual compiler invocation (4) override the others?